template/template.go

```diff
@@ -59,34 +58,13 @@ func (q queryResultByLabelSorter) Swap(i, j int) {
 	q.results[i], q.results[j] = q.results[j], q.results[i]
 }
 
-func query(ctx context.Context, q string, ts time.Time, queryEngine *promql.Engine) (queryResult, error) {
-	query, err := queryEngine.NewInstantQuery(q, ts)
+type QueryFunc func(context.Context, string, time.Time) (promql.Vector, error)
+
+func query(ctx context.Context, q string, ts time.Time, query QueryFunc) (queryResult, error) {
+	vector, err := query(ctx, q, ts)
```

Comment:
can we rename `query` to `queryFn` or `queryFunc`? Looks like recursive call at first glance (: